File: MultiWaveDecoder/jaad/aac/syntax/Element.java
package net.sourceforge.jaad.aac.syntax;

import net.sourceforge.jaad.aac.AACException;
import net.sourceforge.jaad.aac.SampleFrequency;
import net.sourceforge.jaad.aac.sbr.SBR;

public abstract class Element implements Constants {

	private int elementInstanceTag;
	private SBR sbr;

	protected void readElementInstanceTag(BitStream in) throws AACException {
		elementInstanceTag = in.readBits(4);
	}

	public int getElementInstanceTag() {
		return elementInstanceTag;
	}

	void decodeSBR(BitStream in, SampleFrequency sf, int count, bool stereo, bool crc, bool downSampled,bool smallFrames) throws AACException {
		if(sbr==null) sbr = new SBR(smallFrames,elementInstanceTag==ELEMENT_CPE,sf,downSampled);
		sbr.decode(in, count);
	}

	bool isSBRPresent() {
		return sbr!=null;
	}

	SBR getSBR() {
		return sbr;
	}
}

File: MultiWaveDecoder/aac/aac/syntax/Element.cs
// ReSharper disable InconsistentNaming
// ReSharper disable UnusedMember.Global
// ReSharper disable UnusedMember.Local
using System;

namespace MultiWaveDecoder
{
  public abstract class Element : Constants
  {
    int elementInstanceTag;
    SBR sbr;

    protected void readElementInstanceTag(BitStream inStream)
    {
      elementInstanceTag = inStream.readBits(4);
    }

    public int getElementInstanceTag()
    {
      return elementInstanceTag;
    }

    void decodeSBR(BitStream inStream, SampleFrequency sf, int count, bool stereo, bool crc, bool downSampled, bool smallFrames)
    {
      if (sbr == null) sbr = new SBR(smallFrames, elementInstanceTag == ELEMENT_CPE, sf, downSampled);
      throw new NotImplementedException();
      //sbr.decode(inStream, count);
    }

    bool isSBRPresent()
    {
      return sbr != null;
    }

    public SBR getSBR()
    {
      return sbr;
    }
  }
}
